cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1344 @@
+func getSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	path := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		log.Println(path)
+	}
+	return resp, common.SendHTTPGetRequest(path, true, false, &resp)
+}
+
+// CheckExistingExchanges checks if the given exchange exists
+func CheckExistingExchanges(fileName, exchName string) ([]ExchangeInfo, bool, error) {
+	data, err := ReadFileData(fileName)
+	if err != nil {
+		return data, false, err
+	}
+	var resp bool
+	for x := range data {
+		if data[x].Name == exchName {
+			resp = true
+			break
+		}
+	}
+	return data, resp, nil
+}
+
+// CheckMissingExchanges checks if any supported exchanges are missing api checker functionality
+func CheckMissingExchanges(fileName string) ([]string, error) {
+	data, err := ReadFileData(fileName)
+	if err != nil {
+		return nil, err
+	}
+	var tempArray []string
+	for x := range data {
+		tempArray = append(tempArray, data[x].Name)
+	}
+	supportedExchs := exchange.Exchanges
+	for z := 0; z < len(supportedExchs); {
+		if common.StringDataContainsInsensitive(tempArray, supportedExchs[z]) {
+			supportedExchs = append(supportedExchs[:z], supportedExchs[z+1:]...)
+			continue
+		}
+		z++
+	}
+	return supportedExchs, nil
+}
+
+// ReadFileData reads the file data for the json file
+func ReadFileData(fileName string) ([]ExchangeInfo, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+	var data []ExchangeInfo
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+// CheckUpdates checks updates.json for all the existing exchanges
+func CheckUpdates(fileName string) error {
+	var resp []string
+	data, err := ReadFileData(fileName)
+	if err != nil {
+		return err
+	}
+	errMap := make(map[string]error)
+	var wg sync.WaitGroup
+	for x := range data {
+		wg.Add(1)
+		go func(x int) {
+			defer wg.Done()
+			switch data[x].CheckType {
+			case github:
+				sha, err := getSha(data[x].Data.GitHubData.Repo)
+				if err != nil {
+					errMap[data[x].Name] = err
+				}
+				if sha.ShaResp != data[x].Data.GitHubData.Sha {
+					data[x].Data.GitHubData.Sha = sha.ShaResp
+				}
+			case htmlScrape:
+				checkStr, err := CheckChangeLog(*data[x].Data.HTMLData)
+				if err != nil {
+					errMap[data[x].Name] = err
+				}
+				if checkStr != data[x].Data.HTMLData.CheckString {
+					resp = append(resp, data[x].Name)
+					data[x].Data.HTMLData.CheckString = checkStr
+				}
+			}
+		}(x)
+	}
+	wg.Wait()
+	file, err := json.MarshalIndent(data, "", " ")
+	if err != nil {
+		return err
+	}
+	for y := range resp {
+		a, err := GetChecklistItems()
+		if err != nil {
+			return err
+		}
+		for z := range a.CheckItems {
+			if strings.Contains(a.CheckItems[z].Name, resp[y]) {
+				err = UpdateCheckItem(a.CheckItems[z].ID, a.CheckItems[z].Name, a.CheckItems[z].State)
```

Comment:
You should only perform these actions if you have a trello key set. Someone can still use this without utilising trello. Right now the application breaks if you don't have a key set